sdk/core/Azure.Core/src/Response.cs

```diff
@@ -41,6 +41,42 @@ public abstract class Response : IDisposable
         /// </summary>
         public virtual ResponseHeaders Headers => new ResponseHeaders(this);
 
+        private BinaryData? _content;
+
+        /// <summary>
+        /// Gets the contents of HTTP response. Returns a BinaryData with no payload for responses without content.
+        /// </summary>
+        public virtual BinaryData Content
```

Comment:
Are there any concerns here about the case where this method ends up doing IO in the case where the content stream has not been already read into a `MemoryStream`?

Reply:
Yes, I think this property should be assigned in the https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/Internal/ResponseBodyPolicy.cs and throw in the getter when not assigned.

Reply:
I am reading the thread and it seems like we agreed to return empty BD instead of throwing.

Separately, don't like the policy doing two allocations here (BD and Stream). Can policy allocate just one; the other one would be allocated by concrete response? This would allow concrete responses to optimize their storage.

Reply:
Making it nullable would make everyone check it for null if they have nullability enabled. And if we are talking about the same thread it still seems to me we agreed on exception:

You said:
>OK. If we discover that users see this as very problematic, we can always expose Response.IsStreaming property.


>Separately, don't like the policy doing two allocations here (BD and Stream). Can policy allocate just one; the other one would be allocated by concrete response? This would allow concrete responses to optimize their storage.

Yeah, I'm not really a fan of hardcoding a special case for memorystream in Response but maybe it's fine.

Reply:
There was another thread where Matt suggested returning empty and Johan agreed that null vs empty content is not an important distinction.

>Making it nullable would make everyone check it for null if they have nullability enabled. 

Nobody wants to make it nullable.

Reply:
If we were designing the behavior today, I would return empty stream and if we discover somebody needs to know the difference between null and empty, we would add a property to check that.

Reply:
I don't understand. Returning an empty stream when the response is not buffered? How does that help?

Reply:
Sorry, this thread went from null vs empty stream to streaming responses. I was just referring to the "no content" vs empty content distinction.